src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -270,6 +270,11 @@ def validate_dynamic_list_func_response_type(response: Any, f: str):
 
 
 def validate_tool_func_result(func_call_scenario: str, result):
+    if func_call_scenario not in ToolFuncCallScenario.__members__.values():
```

Comment:
What is ToolFuncCallScenario.__members__ ?

Reply:
![image](https://github.com/microsoft/promptflow/assets/20365062/43d958ea-10d0-4bdc-a8cb-fbc1ec702cbd)